src/lib/lib.jl

```diff
@@ -16,7 +16,7 @@ accum(x::Tuple, y::Tuple) = accum.(x, y)
 accum(x::AbstractArray, y::AbstractArray) = accum.(x, y)
 
 @generated function accum(x::NamedTuple, y::NamedTuple)
-  grad(x) = x in fieldnames(y) ? :(y.$x) : :nothing
+  grad(x) = x in fieldnames(y) ? :(y.$x) : :nothing # is this cond ever true? if both x, y are NamedTuples?
```

Comment:
This probably clarifies
```suggestion
  grad(xf) = xf in fieldnames(y) ? :(y.$xf) : :nothing
```

Reply:
ah yes, thanks!